app/scripts/firebase/main.js

```diff
@@ -56,18 +86,52 @@ ArtHop.prototype.signOut = function() {
 ArtHop.prototype.onAuthStateChanged = function(user) {
   if (user) { // User is signed in!
     // Get profile pic and user's name from the Firebase user object.
-
     this.logoutButton.style.display = 'block';
     this.loginButton.style.display = 'none';
     this.nameDisplay.textContent = "Logged in as " + user.displayName;
-  } else { // User is signed out!
-    
+    this.checkForUser(user);
+   } else { // User is signed out!
     this.logoutButton.style.display = 'none';
     this.loginButton.style.display = 'block';
     this.nameDisplay.textContent = '';
   }
 };
 
+ArtHop.prototype.checkForUser = function(user) {
+  if(window.artHop && window.artHop.usersRef !== undefined) {
+    //check if current user is in database
+    window.artHop.usersRef
+      .orderByChild('email')
+      .equalTo(user.email)
+      .once('value')
+      .then(function(data) {
+        if(data.val() === null) {
+          console.log('User is not in database');
+          window.artHop.addUserToDatabase();
+        } else {
+          var currentUser = data.val();
+          window.artHop.rawUser = currentUser;
+          window.artHop.currentUser = currentUser[Object.keys(currentUser)[0]];
+        }
+      })
+      .catch(function(error) {
+        console.log("WTF IS THE ERROR",error)
```

Comment:
Probably need a better error message than this. Did make me laugh though 😆.
I recommend: 

```
console.log("Unknown Check of User Error: ",error)
```